lib/main.coffee

```diff
@@ -188,6 +188,32 @@ module.exports =
         return p
     return execPath
 
+  getProjectPath: (lintedFile) ->
+    projectPaths = atom.project.getPaths()
+    projectNames = projectPaths
+      .map (projectPath) -> (
+        splittedPath = projectPath.split('/')
+        return splittedPath[splittedPath.length - 1]
+      )
+
+    projectName = projectNames
+      .find((projectName) -> lintedFile.indexOf(projectName) > -1)
+
+    return projectPaths[projectNames.indexOf(projectName)]
+
+
+  getExistingConfigFiles: (configFiles, projectPath) ->
+    return configFiles
+      .split(/[ ,]+/)
```

Comment:
Yes, I'll explain myself: this function checks if any of the config files specified on the Atom configuration exists within the project or not. Later, when I push the `--config` parameter, I use the first config file that exists. If none of the configured filenames exists, the `--config` parameter is ignored.

This is the 3d check I do before pushing or not the `--config` parameter

```coffeescript
if (
  (projectConfigFile = atom.config.get('linter-flake8.projectConfigFile')) and
  (projectPath = atom.project.relativizePath(filePath)[0]) and
  (existingConfigFiles = @getExistingConfigFiles(projectConfigFile, projectPath)).length > 0
)
```